### PR TITLE
fix: remove unwanted whitespaces

### LINKED ISSUE #870: Unwanted whitespace before comma
**Describe the bug**

Unwanted white space has been added before comma or point at the end of sentences, as highlighted by the screenshot. 

**Expected behavior**

Comma or point should directly follow the character before.

**Screenshots**

**Additional context**

The problem should be caused by non-properly configured `linter`.

### PR DESCRIPTION
**Pull request by MalinAhlberg**
**Related issue(s) and PR(s)**  
This PR closes #870.

Remove unwanted spaces from html pages, set prettier to not mess with important whitespaces. See **Further comments** sections for more details and discussion.

**Type of change**  

- [X] Bug fix (non-breaking change which fixes an issue)

 
**List of changes made**  

- Removed whitespaces from a bunch of .vue files
- Set `htmlWhitespaceSensitivity` to `css`.

**Screenshot of the fix**  

whitespaces

**Testing**  

Verify that no more unwanted whitespaces are around.
To run prettier (with `ma-exec frontend yarn run format`), make sure that `.prettierrc.json` is updated in the container.

**Further comments**  

`htmlWhitespaceSensitivity` was set to `ignore` in a21f080b80e8b59f661b064710407d0b5b947a34, to stop prettier from splitting html tags. However, this will make prettier treat all whitespaces as non-important. I can't find any way to stop prettier from both introducing whitespaces and splitting tags, it seems like we have to choose. Even though splitted tags are annoying, they are not as annoying as having unwanted whitespaces on the webpage, so I propose having `htmlWhitespaceSensitivity` set to `css`. For more info see [the documentation](https://prettier.io/docs/en/options.html#html-whitespace-sensitivity).

I tried setting `htmlWhitespaceSensitivity` to `strict`, also tried setting  `"bracketSameLine": true`, but no success.

**Definition of Done checklist**  
- [X] My code meets the Acceptance Criteria
- [X] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [X] Tests and lint/format validations are passing
- [X] My changes generate no new warnings

### CODE REVIEW

**Review by nanjiangshu — APPROVED** [review]
Very nice fix! I think it makes sense to set `htmlWhitespaceSensitivity` to `css`. Sometimes we have to accept tradeoffs.

**Review by e0 — APPROVED** [review]
LGTM

Really appreciate you adding the context @MalinAhlberg! I agree with you and @nanjiangshu that this is a good trade off.

**Comment by mihai-sysbio** [comment]
In this [old post](https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting) there is also another option of using `span` and `div` blocks for different results, so I wonder if it is still applicable.

**Comment by MalinAhlberg** [comment]
> In this [old post](https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting) there is also another option of using `span` and `div` blocks for different results, so I wonder if it is still applicable.

Good point! However, it still only provides the two options of either introduced whitespaces or splitted tags (but allows for overriding the value given in the `rc` file).

File: frontend/src/components/Documentation.vue [hunk 1]
```diff
@@ -310,9 +304,8 @@
                 <i>Data overlay</i>
                 sidebar allows uploading of the custom data in
                 <a target="_blank" href="https://en.wikipedia.org/wiki/Tab-separated_values">
-                  TSV format
-                </a>
-                . If the file is parsed correctly, the file name will be highlighted in green; in
+                  TSV format </a
```
Comment (inghylt) [inline frontend/src/components/Documentation.vue:307]: Should we remove the space between `TSV format` and the dot?

File: frontend/src/components/Documentation.vue [hunk 2]
```diff
@@ -571,9 +551,8 @@
             older models that may no longer be maintained (for example HMR 2.0), and others that
             were recently published. The more recent GEMs can also be found in the
             <a href="https://www.github.com/SysBioChalmers/" target="_blank">
-              SysBioChalmers organization GitHub
-            </a>
-            .
+              SysBioChalmers organization GitHub </a
+            >.
```
Comment (inghylt) [inline frontend/src/components/Documentation.vue:555]: Unnecessary extra space

File: frontend/src/components/Documentation.vue [hunk 3]
```diff
@@ -331,9 +324,8 @@
                   target="_blank"
                   href="https://github.com/MetabolicAtlas/data-files/blob/main/DATA_OVERLAY.md#data-source-file-requirements"
                 >
-                  here
-                </a>
-                . The headers of these data series will be shown automatically in the dropdown
+                  here </a
+                >. The headers of these data series will be shown automatically in the dropdown
```
Comment (inghylt) [inline frontend/src/components/Documentation.vue:328]: Unnecessary extra space?

File: frontend/src/components/Documentation.vue [hunk 4]
```diff
@@ -632,10 +611,9 @@
           <p>
             Metabolic Atlas has a
             <a href="/api/v2/" target="_blank">
-              dedicated interface to facilite the use of the API
-            </a>
-            , with output provided in JSON format. When using this service, kindly use a limit of 10
```
Comment (inghylt) [inline frontend/src/components/Documentation.vue:637]: Unnecessary extra space?

File: frontend/src/components/Home.vue [hunk 5]
```diff
@@ -66,9 +66,8 @@
                     Wang, H., et al, 2021.
                     <i>
                       Genome-scale metabolic network reconstruction of model animals as a platform
-                      for translational research
-                    </i>
-                    . PNAS 118, e2102344118
+                      for translational research </i
+                    >. PNAS 118, e2102344118
```
Comment (inghylt) [inline frontend/src/components/Home.vue:70]: Unnecessary extra space?

**Review by inghylt — COMMENTED** [review]
Commented on the places where I found to me a redundant whitespace between a link and a dot. Otherwise it looks good!

**Comment by MalinAhlberg** [comment]
> Commented on the places where I found to me a redundant whitespace between a link and a dot. Otherwise it looks good!

Great that you caught those, @inghylt! Fixed in da96ca5

**Review by inghylt — APPROVED** [review]
Nice work!

**Comment by mihai-sysbio** [comment]
> > In this [old post](https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting) there is also another option of using `span` and `div` blocks for different results, so I wonder if it is still applicable.
> 
> Good point! However, it still only provides the two options of either introduced whitespaces or splitted tags (but allows for overriding the value given in the `rc` file).

Just describe more clarify, I was thinking that if you'd want to keep the previous version of the `rc` file
```
"htmlWhitespaceSensitivity": "ignore"
```
the other alternative would be to change the `<p>` tags to `<span>`. Don't let this comment stop the merge though 😊

**Comment by MalinAhlberg** [comment]
> > > In this [old post](https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting) there is also another option of using `span` and `div` blocks for different results, so I wonder if it is still applicable.
> > 
> > 
> > Good point! However, it still only provides the two options of either introduced whitespaces or splitted tags (but allows for overriding the value given in the `rc` file).
> 
> Just describe more clarify, I was thinking that if you'd want to keep the previous version of the `rc` file
> 
> ```
> "htmlWhitespaceSensitivity": "ignore"
> ```
> 
> the other alternative would be to change the `<p>` tags to `<span>`. Don't let this comment stop the merge though blush

Ah, I see your point now. Let's keep that in mind for future modifications :)

**Pull request merged**